src/main/java/dan200/computercraft/ComputerCraft.java

```diff
@@ -244,6 +245,7 @@
     private static List<IMediaProvider> mediaProviders = new ArrayList<>();
     private static List<ITurtlePermissionProvider> permissionProviders = new ArrayList<>();
     private static final Map<String, IPocketUpgrade> pocketUpgrades = new HashMap<>();
+    private static final List<ILuaAPIFactory> apiFactories = new ArrayList<>();
```

Comment:
By using a `Set` you can avoid explicitly checking if it's been registered later

Reply:
There's an argument that this allows for some level of priority ordering. I'm not entirely sure whether such an ordering is needed though - *ideally* mods shouldn't be overwriting each other's APIs, but it's something worth discussing.

Reply:
I don't think there's any particular reason order is important, but if it is, `LinkedHashSet` maintains order